cpp/src/parquet/encoding.cc

```diff
@@ -2355,6 +2355,83 @@ class DeltaLengthByteArrayDecoder : public DecoderImpl,
   std::shared_ptr<ResizableBuffer> buffered_data_;
 };
 
+// ----------------------------------------------------------------------
+// RLE_BOOLEAN_DECODER
+
+class RleBooleanDecoder : public DecoderImpl,  virtual public BooleanDecoder {
+
+  public:
+    explicit RleBooleanDecoder(const ColumnDescriptor* descr)
+            :DecoderImpl(descr, Encoding::RLE) {}
+
+    void SetData(int num_values, const uint8_t* data, int len) override {
+      num_values_ = num_values;
+      int32_t num_bytes = 0;
+
+      if (len < 4) {
+        throw ParquetException("Received invalid length (corrupt data page?)");
+      }
+      // Load the first 4 bytes, which indicates the legnth
+      num_bytes = ::arrow::util::SafeLoadAs<int32_t>(data);
```

Comment:
`uint32_t`?

https://parquet.apache.org/docs/file-format/data-pages/encodings/#a-namerlearun-length-encoding--bit-packing-hybrid-rle--3

> length := length of the <encoded-data> in bytes stored as 4 bytes little endian (unsigned int32)

Could you parse this as little endian value on both little/big endian environment?

We may be able to use `arrow/util/endian.h` for it...

Reply:
Converted to unsignedint32_t and parsed as little endian.